### PR TITLE
feat(account-lib): implementation of the functionality secretKeyToMne…

### PR DESCRIPTION
**Pull request by JoaquinTissera-Simtlix**
…monic with unit test

implementation of the functionality secretKeyToMnemonic with unit test

[STLX-4676](https://bitgoinc.atlassian.net/browse/STLX-4676)

### CODE REVIEW

File: modules/account-lib/test/unit/coin/algo/utils.ts [hunk 1]
```diff
@@ -21,4 +21,16 @@ describe('utils', () => {
     const objEncodedByAlgo = algosdk.encodeObj(note);
     should.deepEqual(objEncoded, objEncodedByAlgo);
   });
+
+  it('should return a mnemonic from a secret key ', () => {
```
Comment (rliberal-simtlix) [inline modules/account-lib/test/unit/coin/algo/utils.ts:25]: Add test for invalid cases adding an invalid secret key

**Review by rliberal-simtlix — CHANGES_REQUESTED** [review]

File: modules/account-lib/test/unit/coin/algo/utils.ts (hunk 1 above)
Comment (JulioAchaval-SimTLiX) [reply modules/account-lib/test/unit/coin/algo/utils.ts:25]: Same for invalid seed or keyPair, if its possible.

File: modules/account-lib/test/unit/coin/algo/utils.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import algosdk from 'algosdk';
 import utils from '../../../../src/coin/algo/utils';
 import { Algo } from '../../../../src';
 import * as AlgoResources from '../../../resources/algo';
+import { it } from 'mocha';
```
Comment (JulioAchaval-SimTLiX) [inline modules/account-lib/test/unit/coin/algo/utils.ts:6]: Is this necessary?

**Review by JulioAchaval-SimTLiX — COMMENTED** [review]
Please squash your commits after you finish your changes.

File: modules/account-lib/test/unit/coin/algo/utils.ts [hunk 3]
```diff
@@ -21,4 +21,69 @@ describe('utils', () => {
...
+    let mnAlgo;
+    if (keys.prv) {
+      mn = Algo.algoUtils.secretKeyToMnemonic(Buffer.from(keys.prv, 'hex'));
+      mnAlgo = algosdk.secretKeyToMnemonic(Buffer.from(keys.prv, 'hex'));
+    }
+    should.equal(mn, mnAlgo);
+  });
+
+  it('it should return error if the secret key is invalid', () => {
+    const secretKeyInValid = 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa';
```
Comment (rliberal-simtlix) [inline modules/account-lib/test/unit/coin/algo/utils.ts:38]: Adding data like aaaaaaaaaa is valid but not 100% correct.
If you do that is better to do an assertion for a regex. To real validate that is an invalid hex
0[xX][0-9a-fA-F]+

File: modules/account-lib/test/unit/coin/algo/utils.ts [hunk 4]
```diff
@@ -21,4 +21,69 @@ describe('utils', () => {
...
+    const keyPair = Algo.algoUtils.keyPairFromSeed(seed);
+    const { addr, sk } = algosdk.mnemonicToSecretKey(mnemonic);
+    const skOnly = new Uint8Array(Buffer.from(sk.slice(0, 32)));
+    const pk = algosdk.decodeAddress(addr).publicKey;
+    should.deepEqual(keyPair.sk, skOnly);
+    should.deepEqual(keyPair.pk, pk);
+  });
+
+  it('it should return error if the seed is invalid', () => {
+    const seedInValid = new Uint8Array(Buffer.from('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', 'hex'));
```
Comment (rliberal-simtlix) [inline modules/account-lib/test/unit/coin/algo/utils.ts:74]: Same here.

**Review by rliberal-simtlix — CHANGES_REQUESTED** [review]

**Review by rliberal-simtlix — APPROVED** [review]
LGTM :rocket:

**Review by JulioAchaval-SimTLiX — APPROVED** [review]
LGTM

**Review by pancham88 — APPROVED** [review]

**Pull request merged**